### ISSUE TITLE
Refresh token update

### ISSUE THREAD

**Issue by odinuv** [keboola/oauth-api#12]
Component will have two attributes:
- flag - refresh token
- refresh interval

oauth will take care of refreshing the refresh token.
https://github.com/keboola/docker-bundle/issues/221

**Comment by ondrejhlavacek** [comment]
Je nějakej důvod mít refresh interval? Nemohlo by se to refreshovat před každým puštěním jobu? Pak by se muselo zjišťovat, jesti `{last refresh date} + {refresh interval} + {maximum job duration} > {current date}` a to může bejt komplikovaný, kdyby třeba komponenty měly customizovanou délku běhu per projekt/region.

**Comment by odinuv** [comment]
refresh interval je tam pro ten oauth - ten bude sam o sve iniciative token refreshovat v danym intervalu (treba jednou za 14 dni), pred pustenim jobu to nejde, protoze pokud se ten job nepusti treba 14 dni a token vyexpiruje, tak uz nejde refreshnout

**Comment by ondrejhlavacek** [comment]
Jo, tak sorry, to jsem blbě pochopil, jak jsme se tady bavili s Mirem. Refresh interval by teda tohle měl respektovat, abys nedostal token, kterej ti za běhu může expirovat.

**Comment by odinuv** [comment]
jo jasne, mel by to refreshovat s dostatecnym predstihem a ne v posledni vterine :)

**Comment by Halama** [comment]
me tohle porad prijde jako hack, myslim ze by to proste mela vratit komponenta potom co dobehne a docker runner by to jenom ulozil zpet. Tak ma oauth fungovat.

**Comment by odinuv** [comment]
je to hack oauthu, protoze kdyz to udelame jak rikas a nekdo tu komponentu pusti jednou za mesic, tak ji bude muset vzdycky znovu autorizovat pokud ji bude expirovat token za 14 dni

**Comment by Halama** [comment]
jo ale tak by to prave melo bejt. Kdyz ji dlouho nespusti at to expiruje.

**Comment by odinuv** [comment]
no me to prijde v KBC jako dost problem - vidim to jako neco, co samovolne prestava fungovat v zavislosti na tom jak casto to spoustis

**Comment by odinuv** [comment]
+ tohle https://github.com/keboola/docker-bundle/issues/221#issuecomment-347534658

**Comment by MiroCillik** [comment]
Mne sa na tom co sme s Odinom vymysleli v https://github.com/keboola/docker-bundle/issues/221 nepaci len to, ze tam bude musiet bezat nejaky cron, ktory bude kontrolovat tu expiraciu.. 

Ale je pravda, ze pre uzivatela to takto bude lepsie. Hlavne ak mas token co expiruje za 14dni, to je strasne malo. A bude sa to tykat len takychto komponent. 

Ono je to cele vlastne hack, proste sa snazime nejakym trikom vyriesit to, ze niekore OAuth API su navrhnute debilne a maju rychlo expirujuci refresh token.

**Comment by Halama** [comment]
V tom "cronu" zadny problem nevidim. Spis mi proste prijde to reseni divny jako celek, ohybani funkce oauthu.

**Comment by MiroCillik** [comment]
Len si poznacim, ze by sa namiesto "cronu" mohli pouzit CloudWatch eventy.
